src/group/group.controller.ts

```diff
@@ -99,6 +105,8 @@ export class GroupController {
     return this.groupService.update(slug, updateGroupDto);
   }
 
+  @Permissions(UserPermission.DeleteGroups)
```

Comment:
Only `admin` has permission to DeleteGroups. I thought it's gonna be used in global scope allowing admin to perform all possible actions. Each 'group' related endpoint should have a local Group.Member.Permission and above is User.Role.Permission to limit user access to things.

Would it be a good move to make own guard per route instead? Something like
```
import { CanActivate, ExecutionContext, Injectable, ForbiddenException } from '@nestjs/common';
import { Reflector } from '@nestjs/core';

@Injectable()
export class DeleteGroupGuard implements CanActivate {
  constructor(private reflector: Reflector) {}

  async canActivate(context: ExecutionContext): Promise<boolean> {
    const request = context.switchToHttp().getRequest();
    const user = request.user; // User extracted from authentication
    const groupId = request.params.id; // Group ID from route params

    // Check admin-level permission
    if (user.permissions.includes(Permissions.DeleteGroups)) {
      return true;
    }

    // Check ownership and member-level permission
    if (groupMember.permissions.includes(GroupMemberPermission.DeleteGroup)) {
      const group = await Group.findOne({ where: { id: groupId } });
      if (group?.ownerId === user.id) {
        return true;
      }
    }

    throw new ForbiddenException('You do not have permission to delete this group.');
  }
}
```